crates/burn-train/src/metric/classification.rs

```diff
@@ -0,0 +1,26 @@
+use burn_core::prelude::{Backend, Bool, Tensor};
+
+/// Input for classification tasks.
+#[derive(new, Debug, Clone)]
+pub struct ClassificationInput<B: Backend> {
+    /// Sample x Class Non thresholded normalized predictions.
+    pub predictions: Tensor<B, 2>,
+    /// Sample x Class one-hot encoded target.
+    pub targets: Tensor<B, 2, Bool>,
```

Comment:
That's true, I've been looking at the existing code in burn-trai/src/learner/classification and I think it would be easier to just use the adaptor to convert between bin/multiclass/multilable outputs to the general one hot encoded metrics, non?

Reply:
Yeah my comment was not meant to say that this is where is should be handled specifically 😄

Reply:
Yeah of course, I just wanted to get your thoughts on what I had thought to do. Now we have the `ClassificationOutput` and `MultiLabelClassificationOutput` adapted for `ClassificationInput`. It works but I'm still not super happy about it since then the user is able to use, for example `BinaryPrecisionMetric` with a `MultiLabelClassificationOutput`. My idea for the future would be to have separated Inputs and Outputs for each of the classification types such that this would not be possible and would complain at compile time. Thoughts?

Reply:
I understand where you're coming from! But I don't see a straightforward way to do this.. we would have to have a different implementation for the binary, multiclass and multilabel precision metrics because the input is an associated type for the `Metric` trait.

For now I think it's fine.